### PR TITLE
Attribute cleanup

### PR DESCRIPTION
**Pull request by fm-tibco**
Use functions to access attribute data members and do type conversion automatically for attribute values

### CODE REVIEW

**Comment by fm-tibco** [comment]
The only issue I've come across is that mashing is using atttribute.Name in their github.com/TIBCOSoftware/mashling/lib/util/util.go

```
func IsValidTriggerHandlerSetting(metadata *ftrigger.Metadata, property string) bool {
	settings := metadata.Handler.Settings

	for _, element := range settings {
		if element.Name == property {
			return true
		}
	}

	return false
}
```
Will have to change to:

```
func IsValidTriggerHandlerSetting(metadata *ftrigger.Metadata, property string) bool {
	settings := metadata.Handler.Settings

	for _, element := range settings {
		if element.Name() == property {
			return true
		}
	}

	return false
}
```

**Comment by torresashjian** [comment]
@fm-tibco @mellistibco Should we communicate this change before merging? we don't want to break them.

**Comment by mellistibco** [comment]
@jpark800 @jeffreybozek @raji-narayanan please note the change required for this pr.

File: core/data/data.go [hunk 1]
```diff
@@ -48,14 +78,15 @@ func (tv *Attribute) UnmarshalJSON(data []byte) error {
...
 	}
 
-	tv.Name = ser.Name
-	tv.Type, _ = ToTypeEnum(ser.Type)
-	val, err := CoerceToValue(ser.Value, tv.Type)
+	a.name = ser.Name
+	a.dataType, _ = ToTypeEnum(ser.Type)
+	val, err := CoerceToValue(ser.Value, a.dataType)
+
+	return err
```
Comment (mellistibco) [inline core/data/data.go]: should this return be here?

**Pull request merged**